tools/update_setup_version_from_git.sh

```diff
@@ -3,16 +3,20 @@
 #
 # This script is intended to be used before creating a Python package, in order
 # to create a source tarball ("sdist package") with a version which has been
-# defined using the number of git commits since the last tag, without requiring
-# to have the git repository in the package.
+# defined using the number of git commits since the last time setup.py was
+# modified, without requiring to have the git repository in the package.
 #
 # Git tags are expected to be "major.minor", such as "0.1".
 
 set -eu
 cd "$(dirname -- "${BASH_SOURCE[0]}")/.."
 
-LAST_TAG="$(git describe --tags --abbrev=0)"
-COUNT="$(git rev-list --count "${LAST_TAG}..HEAD")"
-VERSION="${LAST_TAG}.${COUNT}"
+VERSION_LINE="$(git grep -h --line-number '^    version=".*",$' setup.py | cut -d : -f 1)"
+LAST_CHANGE="$(git blame --abbrev=40 -L"$VERSION_LINE",+1 HEAD setup.py | cut -d ' ' -f 1)"
+COUNT="$(git rev-list --count "${LAST_CHANGE}..HEAD")"
 
-sed 's/^\( *version=\)"[^"]*"/\1"'"${VERSION}"'"/' -i setup.py
+sed 's/^\(    version=\)"\([0-9]\+\.[0-9]\+\)\.[0-9]\+"/\1"\2.'"$((COUNT - 1))"'"/' -i setup.py
```

Comment:
`COUNT-1` is invalid if `COUNT` is zero. It can happen if the last commit changed `setup.py`.

Reply:
Indeed, I misunderstood why this new algorithm differed by 1 with the current one, but it is been the tag `0.1` was applied on commit d978cadebd2ba006b728dafedce550182b6be3c9, which is the merge commit of the Pull Request which modified the version in `setup.py` in 04b6101100a24e4211a1d16803a3444a6d200402. So `COUNT` should be used as-is. I fixed my code.